### DISCUSSION TITLE
How to download a more specific version of graalvm than "21"?

### DISCUSSION THREAD

**Discussion by borkdude** [oracle/graal#7498] [Q&A] [answered]
I tried `curl -vLO  https://download.oracle.com/graalvm/21/archive/graalvm-jdk-21+35.1_linux-x64_bin.tar.gz` but this doesn't work. What specific versions of 21 are there? I'd like to not change the version of GraalVM from under me in CI.

**Comment by borkdude** [comment]
Perhaps mr. Niephaus knows? @fniephaus :)

**Comment by fniephaus** [comment]
I think you are looking for this, which should point to the 21.0.0 release "forever":
```
https://download.oracle.com/graalvm/21/archive/graalvm-jdk-21_linux-x64_bin.tar.gz
```

For more information, visit: https://www.oracle.com/java/technologies/jdk-script-friendly-urls/

**Reply by borkdude** [reply]
Thanks a lot! I saw that page, but I assumed that just "jdk-21" would automatically upgrade to a new "point" release in the future, e.g. from 21.0.0 to 21.0.2 or whatever?

**Reply by borkdude** [reply]
Ah yes, exactly as that page says:

> These URLs will deliver the current versions of JDK 21 and of GraalVM for JDK 21 with their respective checksums. When 21.0.1 is "current", those URLs deliver 21.0.1. After the release of 21.0.2 in January of 2024 the above URLs will deliver 21.0.2 and so on.

I don't want to have this behavior. My question was about how to get _exactly_ 21.0.0 and always get that version, not 21.0.2 in the future.

**Reply by fniephaus** [reply, marked as answer]
The link I mentioned uses `/archive/` and not `/latest/`, and gives you the GA version and not the latest version.

**Reply by borkdude** [reply]
@fniephaus Got it, thank you!
